react-js/react-part-2.md

```diff
@@ -0,0 +1,184 @@
+# React Part 2
+
+## Projected Time
+
+1.5 hours
+- Lesson: 30min
+- Guided Practice: 30min
+- Independent Practice: 30min
+
+### Prerequisites
+
+- [React](react.md)
+- [JSX](react.md)
+- [Props](react.md)
+
+### Motivation
+
+In introduction to [React](/react.md) we discussed about how React makes websites interactive and reduces jQuery code. Now the problem arises if we want to store some information which intent to change and due to change of that information the UI changes i.e the page is re-rendered. Thus, component state is used for such cases.
+
+### Objectives
+
+- Component state
+- State vs Props
+- State are Immutable
+
+### Lesson
+
+- [Introduction to component state in React (video)](https://www.youtube.com/watch?v=e5n9j9n83OM)
+- [Component state in React slides](https://docs.google.com/presentation/d/1Trj_7vWYe79_mDboDxN9rdT94yjY5Dt55xtU1k3dUsw/edit?usp=sharing)
+
+### Common Mistakes & Misconceptions
+
+1. The scope of the state is local to the component.
+2. Don't forget to update state immutably.
+
+### Guided Practice
+
+1. Define state.
+  - >	``` state = {} ```
+2. Initialize state
+  - State in class component can be initialized in two ways
+    - Directly inside the 
+
```

Comment:
This is unfinished. Also, I don't understand why you have No. 1 and No. 2 

No. 2 should cover it. Add a code snippet showing this.state initialization in a constructor (with at least one key and value)

Reply:
@Kundan28 can you review more carefully so you don't approve this type of content.

Reply:
Okay, I'll do the changes.